src/EnergyPlus/BoilerSteam.hh

```diff
@@ -70,57 +70,50 @@ namespace BoilerSteam {
+        int BoilerOutletNodeNum = 0;                // Node number at the boiler outlet
+        std::array<Real64, 3> FullLoadCoef = {0.0}; // Coefficients of the fuel consumption/part load ratio curve
+        int TypeNum = 0;                            // Plant loop type identifier
         PlantLocation plantLoc;
-        int PressErrIndex;             // index pointer for recurring errors
-        int FluidIndex;                // Steam index
+        int PressErrIndex = 0;         // index pointer for recurring errors
+        int FluidIndex = 0;            // Steam index
         std::string EndUseSubcategory; // identifier use for the end use subcategory
-        bool myFlag;
-        bool myEnvrnFlag;
+        bool myFlag = true;
+        bool myEnvrnFlag = true;
 
-        Real64 FuelUsed;           // W - Boiler fuel used
-        Real64 BoilerLoad;         // W - Boiler Load
-        Real64 BoilerEff;          // Boiler efficiency
-        Real64 BoilerMassFlowRate; // kg/s - Boiler mass flow rate
-        Real64 BoilerOutletTemp;   // W - Boiler outlet temperature
+        Real64 FuelUsed = 0.0;           // W - Boiler fuel used
+        Real64 BoilerLoad = 0.0;         // W - Boiler Load
+        Real64 BoilerEff = 0.0;          // Boiler efficiency
+        Real64 BoilerMassFlowRate = 0.0; // kg/s - Boiler mass flow rate
+        Real64 BoilerOutletTemp = 0.0;   // W - Boiler outlet temperature
 
-        Real64 BoilerEnergy;    // J - Boiler energy integrated over time
-        Real64 FuelConsumed;    // J - Boiler Fuel consumed integrated over time
-        Real64 BoilerInletTemp; // C - Boiler inlet temperature
+        Real64 BoilerEnergy = 0.0;    // J - Boiler energy integrated over time
+        Real64 FuelConsumed = 0.0;    // J - Boiler Fuel consumed integrated over time
+        Real64 BoilerInletTemp = 0.0; // C - Boiler inlet temperature
 
-        std::string BoilerFuelTypeForOutputVariable;
+        std::string BoilerFuelTypeForOutputVariable = "";
 
         // Default Constructor
-        BoilerSpecs()
-            : FuelType(DataGlobalConstants::ResourceType::None), Available(false), ON(false), MissingSetPointErrDone(false), UseLoopSetPoint(false),
-              DesMassFlowRate(0.0), MassFlowRate(0.0), NomCap(0.0), NomCapWasAutoSized(false), NomEffic(0.0), MinPartLoadRat(0.0),
-              MaxPartLoadRat(0.0), OptPartLoadRat(0.0), OperPartLoadRat(0.0), TempUpLimitBoilerOut(0.0), BoilerMaxOperPress(0.0),
-              BoilerPressCheck(0.0), SizFac(0.0), BoilerInletNodeNum(0), BoilerOutletNodeNum(0), FullLoadCoef(3, 0.0), TypeNum(0), plantLoc{},
-              PressErrIndex(0), FluidIndex(0), myFlag(true), myEnvrnFlag(true), FuelUsed(0.0), BoilerLoad(0.0), BoilerEff(0.0),
-              BoilerMassFlowRate(0.0), BoilerOutletTemp(0.0), BoilerEnergy(0.0), FuelConsumed(0.0), BoilerInletTemp(0.0),
-              BoilerFuelTypeForOutputVariable("")
+        BoilerSpecs() : plantLoc{}
```

Comment:
Why is this necessary? Why is `plantLoc` default constructor not called automatically?

Reply:
this was part of a diff debug and has been removed